### PR TITLE
zig: build offline documentation

### PR DESCRIPTION
**Pull request by KAction**
maintainers: @aiotter @andrewrk @AndersonTorres

### CODE REVIEW

**Comment by AndersonTorres** [comment]
This is a good opportunity to use multiple outputs!

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/zig/0.10.nix [hunk 1]
```diff
@@ -52,7 +53,18 @@ stdenv.mkDerivation rec {
     "-DZIG_TARGET_MCPU=baseline"
   ];
 
+  postBuild = ''
+    ln -sf ../doc .
+    ./zig2 build-exe ./doc/docgen.zig
+    ./docgen ./zig2 ./doc/langref.html.in ./langref.html
+  '';
```
Comment (AndersonTorres) [inline pkgs/development/compilers/zig/0.10.nix:60]:
```suggestion
  postBuild = ''
    ./zig2 build-exe ../doc/docgen.zig
    ./docgen ./zig2 ../doc/langref.html.in ./langref.html
  '';
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by KAction** [comment]
Applied suggestions.

**Comment by AndersonTorres** [comment]
Does it work on Zig 0.9?

**Comment by KAction** [comment]
Not verbatim:
```
./zig2: No such file or directory
```

What about we merge it for latest version?

**Comment by AndersonTorres** [comment]
OK for now.

**Pull request merged**